### PR TITLE
chore(protocol_tests): updates for latest protocol tests

### PR DESCRIPTION
**Pull request by kstich**
This PR will be paired with one to [awslabs/smithy-typescript](https://github.com/awslabs/smithy-typescript): https://github.com/awslabs/smithy-typescript/pull/246

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

-----

### chore: update codegen plugin version and deps

-----

### fix: fix issues in query and xml serde

This commit fixes issues regarding the `@sparse` trait and null
structure value serialization in the awsQuery, ec2Query, and restXml
AWS protocols.

-----

### fix: several issues in JSON protocol serde generation

This commit fixes several issues when generating code for JSON
related AWS protocols discovered when incorporating the newest AWS
protocol tests. It contains the following fixes:

* Fixes issues around the handling of the `@sparse` trait.
* Fixes issues in serializing null values of structure members.
* Fixes issues loading error codes for RPC JSON protocols.
* Fixes deserializing document type payload responses.

-----

### chore: update generated protocol tests

-----

### fix: fix issues in some service models

### CODE REVIEW

**Comment by kstich** [comment]
Taking a look at conflicting files.

File: protocol_tests/aws-restxml/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@aws-sdk/aws-restxml",
   "description": "@aws-sdk/aws-restxml client",
-  "version": "1.0.0-rc.7",
+  "version": "1.0.0-alpha.1",
```
Comment (mtdowling) [inline protocol_tests/aws-restxml/package.json:4]: is this right?

**Review by mtdowling — APPROVED** [review]

**Comment by kstich** [comment]
Conflicting file issues have been resolved. CI failing because of codegen package version change which will need awslabs/smithy-typescript#246 merged to succeed.

**Review by trivikr — APPROVED** [review]
Verified that Java CI is successful when run locally

**Pull request merged**